### PR TITLE
signal-cli: enable dbus functionality

### LINKED ISSUE #72737: `signal-cli daemon` not supported
`signal-cli -u <user> daemon` fails on 19.09 and 2019-11-03's unstable with the following error:
```
Missing native library dependency for dbus service: no unix-java in java.library.path
```

**To Reproduce**:
1. Install `signal-cli` from `nixos-19.09` or `nixos-unstable`.
2. Register a user.
3. Invoke `signal-cli -u +19738002144`.

This blocks https://github.com/NixOS/nixpkgs/issues/62102.

### PR DESCRIPTION
**Pull request by erictapen**
###### Motivation for this change

Close #72737
 
###### Things done

Refined the code from https://github.com/NixOS/nixpkgs/issues/72737#issuecomment-549170143.

Now the following should work:

```
# Starting the daemon
./result/bin/signal-cli -u MY_NUMBER daemon
```

```
# Sending messages while the daemon is running
./result/bin/signal-cli --dbus send -m "test message" RECIPIENT_NUMBER
```

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @ivan

### CODE REVIEW

**Comment by erictapen** [comment]
@GrahamcOfBorg build signal-cli

**Review by ivan — APPROVED** [review]
This all looks fine and I confirmed that it works, at least after I set `DBUS_SESSION_BUS_ADDRESS` (not present in my SSH session).

Since you're using and maintaining this more than I am, would you also like to add yourself or a co-author as a maintainer?

**Comment by erictapen** [comment]
> Since you're using and maintaining this more than I am, would you also like to add yourself or a co-author as a maintainer?

Sure!

**Review by ivan — APPROVED** [review]

**Review by ivan — CHANGES_REQUESTED** [review]
Actually, I just noticed that `dbus_java` and `libmatthew_java` are Linux-only, it would be good to keep this working on darwin in the spirit of not breaking things that worked previously.

**Comment by erictapen** [comment]
Personally I don't think I'd need to fix the breakage for darwin, as

- we don't know wether anyone on darwin actually uses this
- I don't have the tools to test wether this actually builds on darwin.

But as this one is cheap I added some extra logic that hopefully enables the darwin package to still build.

**Comment by Mic92** [comment]
@GrahamcOfBorg build signal-cli

File: pkgs/applications/networking/instant-messengers/signal-cli/default.nix [hunk 1]
```diff
@@ -10,16 +10,24 @@ stdenv.mkDerivation rec {
     sha256 = "1gvdifscyxmxn2dwlkqi684ahy5kbcj84mqda0m8l4aa8iaq1d59";
   };
 
-  buildInputs = [ makeWrapper ];
+  buildInputs = [ libmatthew_java dbus dbus_java ];
```
Comment (Mic92) [inline pkgs/applications/networking/instant-messengers/signal-cli/default.nix]: This should fix evaluation on macOS:
```suggestion
```

It is not required on Linux as far as I can see?

Comment (erictapen) [reply pkgs/applications/networking/instant-messengers/signal-cli/default.nix]: Yeah I somehow forgot to commit that one. Is fixed.

**Comment by erictapen** [comment]
@GrahamcOfBorg build signal-cli

**Review by ivan — APPROVED** [review]
Thanks! This looks fine and I confirmed that it still builds on macOS 10.14.

**Review by 0x4A6F — APPROVED** [review]
Nice, so with a slightly modified #72948 (with https://github.com/jwoglom/npyscreen/commit/d01ce55df23ff1596ff874f20b85a1270f42b015) signal-ncurses can utilize signal-cli.

**Comment by erictapen** [comment]
So anybody wants to merge this? I guess this is low hanging fruit, as it got approved by two people.

**Pull request merged**